### PR TITLE
Sanitize db name for dacpac/bacpac file names

### LINKED ISSUE #5216: Sanitize filenames that could have invalid characters from db name
Database names can have characters that are invalid in file names, and since both the dacpac extension and schema compare extension use the database name to generate default file names, sanitation needs to be added.

Steps to Reproduce:

1. Create a db with an invalid file name character like '<'
2. Try to extract dacpac
3. Extract fails with error

image

### PR DESCRIPTION
**Pull request by kisantia**
Database names can have characters that are invalid in filenames, so this replaces those invalid characters with an underscore. For example, in|valid*<db?name would become in_valid__db_name. This also adds a few unit tests to test this function.

Fixes #5216

### CODE REVIEW

File: extensions/dacpac/src/wizard/api/utils.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+const INVALID_FILE_CHARS_Windows = /[\\/:\*\?"<>\|]/g;
+const INVALID_FILE_CHARS = /[\\/]/g;
+
+/**
+ * Determines if a given character is a valid filename character
+ * @param c Character to validate
+ */
+export function isValidFilenameCharacter(c: string): boolean {
```
Comment (alanrenmsft) [inline extensions/dacpac/src/wizard/api/utils.ts]: probably there is no need to export this function

File: extensions/dacpac/src/test/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+const path = require('path');
+import * as testRunner from 'vscodetestcover';
+
+const suite = 'DacFx Tests';
+
+const testOptions: any = {
+	ui: 'bdd',
+	useColors: true,
+	timeout: 600000
```
Comment (alanrenmsft) [inline extensions/dacpac/src/test/index.ts]: do you really need the timeout to be so long? 60 seconds should be more than enough.

File: extensions/dacpac/src/test/index.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+const coverageConfig: any = {
+	coverConfig: '../../coverageConfig.json'
```
Comment (Charles-Gagnon) [inline extensions/dacpac/src/test/index.ts]: I don't see the coverageConfig.json in here

File: extensions/dacpac/src/test/dacpac.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	it('Should determine invalid Windows file name characters', async () => {
+		let isWindows = os.platform() === 'win32';
+
+		// invalid only for Windows
+		should(isValidFilenameCharacter('?')).equal(isWindows ? false : true);
+		should(isValidFilenameCharacter(':')).equal(isWindows ? false : true);
+		should(isValidFilenameCharacter('*')).equal(isWindows ? false : true);
+		should(isValidFilenameCharacter('<')).equal(isWindows ? false : true);
+		should(isValidFilenameCharacter('>')).equal(isWindows ? false : true);
+		should(isValidFilenameCharacter('|')).equal(isWindows ? false : true);
```
Comment (Charles-Gagnon) [inline extensions/dacpac/src/test/dacpac.test.ts]: Missing "

File: extensions/dacpac/src/test/index.ts (hunk 3 above)
Comment (kisantia) [reply extensions/dacpac/src/test/index.ts]: added

**Pull request merged**